src/sentry/api/endpoints/monitor_checkins.py

```diff
@@ -140,6 +149,21 @@ def post(
         result = serializer.validated_data
 
         with transaction.atomic():
+            environment_name = result.get("environment")
+            if not environment_name:
```

Comment:
https://github.com/getsentry/sentry/pull/45192#discussion_r1122476220
Not sure I want to add a default here in case request passes `null`. Need to discuss later